scalameta/contrib/src/main/scala/scala/meta/contrib/ScaladocParser.scala

```diff
@@ -0,0 +1,120 @@
+package scala.meta.contrib
+
+import DocToken._
+import scala.meta.Syntax
+import scala.meta.tokens.Token.Comment
+
+object ScaladocParser {
+
+  /**
+    * Parses a scaladoc [[Comment]].
+    */
+  def parseScaladoc(comment: Comment): Seq[DocToken] = {
+
+    val trimmedComment: String = comment.show[Syntax].trim
+
+    require(
+      requirement = trimmedComment.startsWith("/*") && trimmedComment.endsWith("*/"),
+      message = "Input comment is not a Scaladoc"
+    )
+
+    val scaladocSymbols = Seq('/', '*', ' ')
+
+    val lineSeparedDocString: Seq[String] =
+      trimmedComment
+        .split(Array('\n', '\r')) // Splits the Scaladoc in lines
```

Comment:
@olafurpg thank you a lot for your help, now I am getting more familiar with fastparser, I should have known this tool before, it is amazing :)

This is the final parser, it ignores also the code from the beginning, previous to the code block insertion, and it captures the start/end of the code block (For removing the codeblock from the code to parse):

```scala
case class PositiveCodeBlockParserOutput(beginIndex: Int,
                                         codeBlock: DocToken,
                                         endIndex: Int)

val codeBlockParser = {
   P(
      // Code before the code block (ignored)
      ((AnyChar ~ !"{{{").rep ~ AnyChar).?
      // Captures the index where the code block begins.
      ~ Index
      // Code block start
      ~ "{{{"
      // Code within the code block.
      ~ ((AnyChar ~ !"}}}").rep ~ AnyChar).!.map(c => DocToken(CodeBlock, c.trim))
      // Captures the index where the code block ends.
      ~ Index
      // Code block end
      ~ "}}}"
  ).map(PositiveCodeBlockParserOutput.tupled)
}
```